scrapers/scraper.py

```
@@ -0,0 +1,424 @@
+            if "badges/" in url:
+                parts = url.split("/")
+                
+                for p in parts:
+                    if p.isdigit():
+                        lines[name][type] = int(p)
+                        break
+                break
+    except:
+        pass
+
+def add_badges(url_id, badge_type="badge", x=None, custom={}):
```

Ruff (0.12.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)